db/crud.go

```diff
@@ -869,6 +871,42 @@ func (db *DatabaseCollectionWithUser) OnDemandImportForWrite(ctx context.Context
 	return nil
 }
 
+// updateMutateInSpec returns the mutate in spec needed for the document update based off the outcome in updateHLV
+func updateMutateInSpec(hlv HybridLogicalVector) []sgbucket.MacroExpansionSpec {
+	var outputSpec []sgbucket.MacroExpansionSpec
+	if hlv.Version == math.MaxUint64 {
+		spec := sgbucket.NewMacroExpansionSpec(xattrCurrentVersionPath(base.SyncXattrName), sgbucket.MacroCas)
+		outputSpec = append(outputSpec, spec)
+	}
+	if hlv.CurrentVersionCAS == math.MaxUint64 {
+		spec := sgbucket.NewMacroExpansionSpec(xattrCurrentVersionCASPath(base.SyncXattrName), sgbucket.MacroCas)
+		outputSpec = append(outputSpec, spec)
+	}
+	return outputSpec
+}
+
+// updateHLV updates the HLV in the sync data appropriately based on what type of document update event we are encountering
+func (db *DatabaseCollectionWithUser) updateHLV(d *Document, docUpdateEvent DocUpdateEvent) (*Document, error) {
+
+	switch docUpdateEvent.eventType {
+	case BlipWriteEvent:
+		// preserve any other logic on the HLV that has been done by the client, only update to cvCAS will be needed
+		d.HLV.CurrentVersionCAS = hlvExpandMacroCASValue
+	case ImportEvent:
+		// VV should remain unchanged
+	case SGWriteEvent:
+		// add a new entry to the version vector
+		newVVEntry := CurrentVersionVector{}
+		newVVEntry.SourceID = db.dbCtx.BucketUUID
+		newVVEntry.VersionCAS = hlvExpandMacroCASValue
```

Comment:
I think the current XDCR doc suggests populating cvCAS on an SG write as well.

Reply:
Updated to reflect this, also added assertions to relevant test to reflect this change.